### PR TITLE
Update docs to reflect web SDK v4 changes

### PR DESCRIPTION
**Pull request by kochis**
Update docs to reflect Web SDK V4 changes.

https://github.com/radarlabs/radar-sdk-js/pull/104

### CODE REVIEW

File: docs/maps/maps.mdx [hunk 1]
```diff
@@ -95,35 +64,39 @@ To create a simple web page with a map:
 
 To create a [React](https://react.dev) component with a map:
 
+<Alert alertType='info'>
+  The radar-js-sdk has a depdnency on maplibre-gl for the map rendering.
```
Comment (lmeier) [inline docs/maps/maps.mdx:68]: sp: dependency

Comment (lmeier) [inline docs/maps/maps.mdx:68]: sp: dependency

File: docs/maps/maps.mdx [hunk 2]
```diff
@@ -95,35 +64,39 @@ To create a simple web page with a map:
 
 To create a [React](https://react.dev) component with a map:
 
+<Alert alertType='info'>
+  The radar-js-sdk has a dependency on maplibre-gl for the map rendering.
```
Comment (brettguenther) [inline docs/maps/maps.mdx:68]: The radar-sdk-js

File: docs/maps/maps.mdx [hunk 3]
```diff
@@ -165,17 +131,19 @@ export default {
     const map = shallowRef(null);
 
     onMounted(() => {
-      map.value = markRaw(new maplibregl.Map({
+      Radar.initialize(PUBLISHABLE_KEY);
```
Comment (brettguenther) [inline docs/maps/maps.mdx:134]: should keep publishable key value/variable syntax consistent.  'prj_live_pk_...' is used later in the doc

File: docs/sdk/web.mdx [hunk 4]
```diff
@@ -302,13 +365,25 @@ Radar.getMatrix({
...
+  });
+})
+.catch((err) => {
+  // handle error
 });
 ```
 
+## Maps & UI Kits
+
+In addition to APIs, the Radar Web SDK has built-in UI components for building rich user experiences powere by Radar.
```
Comment (brettguenther) [inline docs/sdk/web.mdx:383]: powered

File: cspell.json [hunk 5]
```diff
@@ -0,0 +1,64 @@
...
+    "objc",
+    "onesignal",
+    "podfile",
+    "pubspec",
+    "quickstart",
+    "quickstarts",
+    "radarlabs",
+    "sfmc",
+    "signup",
+    "ucec",
```
Comment (jsani-radar) [inline cspell.json:54]: what's this one?

File: cspell.json [hunk 6]
```diff
@@ -0,0 +1,64 @@
...
+    "upsert",
+    "upserted",
+    "upserts",
+    "xcframework",
+    "xcodeworkspace",
+    "xcproject",
+    "xcworkspace",
+    "yxssullpy"
```
Comment (jsani-radar) [inline cspell.json:62]: ditto

File: cspell.json [hunk 7]
```diff
@@ -0,0 +1,64 @@
...
+  "ignorePaths": [
+    "**/docs/places/*",
+    "**/docs/regions/*",
+    "**/docs/places.mdx"
+  ],
+  "ignoreWords": [
+    "androidx",
+    "appboy",
+    "autocompletes",
+    "bkjcium",
```
Comment (jsani-radar) [inline cspell.json:16]: this one too

File: cspell.json [hunk 8]
```diff
@@ -0,0 +1,64 @@
...
+    "blocklisted",
+    "brandify",
+    "cartfile",
+    "ccpa",
+    "cgeq",
```
Comment (jsani-radar) [inline cspell.json:21]: should be the last one

File: docs/sdk/web.mdx [hunk 9]
```diff
@@ -3,51 +3,80 @@ sidebar_label: Web
...
+<link href="https://js.radar.com/v4.0.0/radar.css" rel="stylesheet">
+<script src="https://js.radar.com/v4.0.0/radar.min.js"></script>
 ```
 
-In a web app, install the package from npm, then import the library:
+In a web app, install the package from npm:
 
 ```bash
-npm install --save radar-sdk-js
+npm install --save radar-sdk-js maplibre-gl
```
Comment (jsani-radar) [inline docs/sdk/web.mdx:24]: hmm wondering if this should be the default instruction? what if someone wants to power a non-maps use case with the web SDK?

File: docs/sdk/web.mdx [hunk 10]
```diff
@@ -3,51 +3,80 @@ sidebar_label: Web
...
 ```
+<Alert alertType='info'>
+  The Radar SDK has a dependency on maplibre-gl for map rendering.
+</Alert>
```
Comment (jsani-radar) [inline docs/sdk/web.mdx:28]: perhaps better to put the `npm install --save maplibre-gl` instruction here to separate it out

File: docs/sdk/web.mdx [hunk 11]
```diff
@@ -3,51 +3,80 @@ sidebar_label: Web
...
 
+Then import the library and corresponding stylesheet:
 ```javascript
 import Radar from 'radar-sdk-js';
-```
 
-The SDK is less than 20 KB minified.
+import 'radar-sdk-js/dist/radar.css';
```
Comment (jsani-radar) [inline docs/sdk/web.mdx:34]: ditto, should we callout that this is only for maps?

File: docs/sdk/web.mdx [hunk 12]
```diff
@@ -85,112 +114,130 @@ The SDK uses the [HTML5 geolocation API](https://developer.mozilla.org/en-US/doc
...
+  const { location, user, events } = result;
+  // do something with location, user, or events
+})
+.catch((err) => {
+  // handle error
 });
 ```
 
-#### Context
+### Get Context
```
Comment (jsani-radar) [inline docs/sdk/web.mdx:180]: fwiw we call this `Context` in all the other SDK refs

File: docs/sdk/web.mdx [hunk 13]
```diff
@@ -234,34 +287,41 @@ Radar.searchPlaces({
...
   postalCode: "10003",
   countryCode: "US",
-}, function(err, { address, result }) {
-  if (!err) {
-    // do something with address and validation result
-  }
+})
+.then((result) => {
+  const { address, result } = result;
+  // do something with result
```
Comment (jsani-radar) [inline docs/sdk/web.mdx:312]: should we rename the var for `result.result`? could be confusing with the callback param

File: docs/sdk/web.mdx [hunk 14]
```diff
@@ -275,19 +335,22 @@ Radar.getDistance({
...
   ],
   units: 'imperial'
-}, function(err, result) {
-  if (!err) {
-    // do something with result.routes
-  }
+})
+.then((result) => {
+  const { routes } = result;
+  const { geodesic, foot, bike, car } = routes;
```
Comment (jsani-radar) [inline docs/sdk/web.mdx:341]: add `// do something with distance result` to be consistent

File: cspell.json [hunk 15]
```diff
@@ -0,0 +1,64 @@
+{
+  "version": "0.2",
+  "language": "en",
+  "showSuggestions": true,
+  "unique": true,
+  "silent": true,
+  "ignorePaths": [
+    "**/docs/places/*",
+    "**/docs/regions/*",
+    "**/docs/places.mdx"
```
Comment (jsani-radar) [inline cspell.json:10]: why ignoring the Places Overview page?

File: docs/sdk/web.mdx [hunk 16]
```diff
@@ -3,51 +3,80 @@ sidebar_label: Web
...
-provide a `cacheLocationMinutes` value as an option when initializing:
 ```javascript
-Radar.initialize(publishableKey, {
-  cacheLocationMinutes: 30
+Radar.initialize('prj_live_pk_...', {
+  /* custom options here */
 });
 ```
 
+The available options are:
```
Comment (jsani-radar) [inline docs/sdk/web.mdx:61]: this is a solid addition, thanks for fleshing this out

**Review by jsani-radar — COMMENTED** [review]
looking good overall, just some small comments

File: cspell.json (hunk 8 above)
Comment (kochis) [reply cspell.json:24]: These were all fraud related, but now realizing the aren't "words" on the page. Can probably remove these, but need to figure out why they were flagged as misspellings.

Comment (jsani-radar) [reply cspell.json:24]: not a big deal, just seemed strange that they'd be necessary - saw you added a few more too

File: docs/sdk/web.mdx (hunk 9 above)
Comment (kochis) [reply docs/sdk/web.mdx:24]: It's not possible, it will throw an error if the dependency is not included.

File: docs/maps/maps.mdx [hunk 17]
```diff
@@ -206,19 +174,27 @@ body {
...
+In addition, you can specify the Radar theme with the `style` property:
+
+<div class="api-params-table">
+
+| Name  | Default | Possible Values | Description           |
+|-------|---------|-----------------|-----------------------|
+|`style` |`radar-default-v1` |`radar-default-v1` |The Radar map theme used for the map style. |
+
+<Alert alertType='tip'>
+  To center the map based on the user's location, create the map without the <code>center</code> and <code>zoom</code> properties. <br />The initial map position will be based on the user's <a href="/documentation/api#ip-geocode">IP Geolocation</a>.
```
Comment (palmakkar) [inline docs/maps/maps.mdx:193]: nit: feels a little wordy for a tip - can we shorten to something like "The map will be automatically centered based on the user's IP Geolocation unless center and zoom properties are specified."?

File: docs/maps/maps.mdx [hunk 18]
```diff
@@ -206,19 +174,27 @@ body {
...
+
+When creating a new Radar map, you can configure it with any of the [MapOptions](https://maplibre.org/maplibre-gl-js/docs/API/types/maplibregl.MapOptions/) available in MapLibre.
+
+In addition, you can specify the Radar theme with the `style` property:
+
+<div class="api-params-table">
+
+| Name  | Default | Possible Values | Description           |
+|-------|---------|-----------------|-----------------------|
+|`style` |`radar-default-v1` |`radar-default-v1` |The Radar map theme used for the map style. |
```
Comment (palmakkar) [inline docs/maps/maps.mdx:190]: do we need this? just thinking it looks a little odd since we only have one and it's the default

**Review by palmakkar — COMMENTED** [review]
looks great - leaving a few nits

File: docs/maps/maps.mdx (hunk 18 above)
Comment (kochis) [reply docs/maps/maps.mdx:190]: Yeah, we can omit for now. Maybe bring it back when we add the additional themes.

**Pull request merged**